braindecode/models/base.py

```diff
@@ -0,0 +1,112 @@
+from typing import List
+
+
+class EEGModuleMixin:
+    """
+    Mixin class for EEG models.
+
+    Parameters
+    ----------
+    n_channels: int
+        Number of EEG channels.
+    channel_names: list of str
+        Names of the EEG channels.
+    input_window_samples: int
+        Number of time samples of the input window.
+    input_window_seconds: float
+        Length of the input window in seconds.
+    sfreq: float
+        Sampling frequency of the EEG recordings.
+
+    .. note::
+
+       If some input signal-related parameters are not specified,
+       there will be an attempt to infer them from the other parameters.
+
+    Raises
+    ------
+    AttributeError: If some input signal-related parameters are not specified and can not be inferred.
+    """
+
+    def __init__(
+            self,
+            n_channels: int = None,
+            channel_names: List[str] = None,
+            input_window_samples: int = None,
```

Comment:
while we're at it I would rename it to n_times also to be consistent with mne but it's a matter of taste 

cc @robintibor